### ISSUE TITLE
FileDialog.SetupSideBar hangs when DriveInfo.GetDrives blocks waiting for an unavailable network drive

### ISSUE THREAD

**Issue by DeaLumi** [goatcorp/Dalamud#1191]
Hiyo! This is probably a fairly fringe issue, but per the title, `DriveInfo.GetDrives()` (called [here](https://github.com/goatcorp/Dalamud/blob/master/Dalamud/Interface/ImGuiFileDialog/FileDialog.Files.cs#L301)) is a blocking call, and can take 30+ seconds to complete in my very particular case.

For reference, my very particular case is as such: My home network includes a "NAS" (converted from an old computer) which is mapped as a network drive and assigned a drive letter (N:) in Windows. I usually leave the NAS powered down, since it uses an old PSU and likely isn't terribly efficient. The net result is that I generally have a drive listed which can't actually be reached -- and it turns out (also noted on [this stackoverflow post](https://stackoverflow.com/questions/1612408/system-io-driveinfo-getdrives-hangs-with-disconnected-network-drives)) that this causes `GetDrives()` to hang for a while before eventually omitting that drive from its results.

- I've only had this crash the game once, when a plugin hit some kind of concurrent timeout, so I don't think it's a serious stability threat.
- I have workarounds available (leaving the NAS on while I play FF, or removing the network drive mapping).
- I think a fix is fairly complicated -- the naive answer is "do it in a different thread", but I'm not familiar with the threading model the UI uses/ability to add drives well after the the UI is displayed/etc.

**Comment by goaaats** [comment]
This probably duplicates #1130, do you mind posting your research there?

**Comment by DeaLumi** [comment]
Yup, it absolutely does; my apologies. (I promise I first searched for "file" and "hang" before starting a new issue...) I'll add a comment there, but it looks like a fix is already planned.

Sorry again for the needless issue!
